src/index.ts

```diff
@@ -0,0 +1,489 @@
+    return new Wallet(seed)
+  }
+
+  public static fromV3(
+    input: string | V3Keystore,
+    password: string,
+    nonStrict: boolean = false,
+  ): Wallet {
+    const json: V3Keystore =
+      typeof input === 'object' ? input : JSON.parse(nonStrict ? input.toLowerCase() : input)
+
+    if (json.version !== 3) {
+      throw new Error('Not a V3 wallet')
+    }
+
+    let derivedKey: Buffer, kdfparams: any
+    if (json.crypto.kdf === 'scrypt') {
+      kdfparams = json.crypto.kdfparams
+
+      // FIXME: support progress reporting callback
+      derivedKey = scryptsy(
+        Buffer.from(password),
+        Buffer.from(kdfparams.salt, 'hex'),
+        kdfparams.n,
+        kdfparams.r,
+        kdfparams.p,
+        kdfparams.dklen,
+      )
+    } else if (json.crypto.kdf === 'pbkdf2') {
+      kdfparams = json.crypto.kdfparams
+
+      if (kdfparams.prf !== 'hmac-sha256') {
+        throw new Error('Unsupported parameters to PBKDF2')
+      }
+
+      derivedKey = crypto.pbkdf2Sync(
+        Buffer.from(password),
+        Buffer.from(kdfparams.salt, 'hex'),
+        kdfparams.c,
+        kdfparams.dklen,
+        'sha256',
+      )
+    } else {
+      throw new Error('Unsupported key derivation scheme')
+    }
+
+    const ciphertext = Buffer.from(json.crypto.ciphertext, 'hex')
+    const mac = ethUtil.keccak256(Buffer.concat([derivedKey.slice(16, 32), ciphertext]))
+    if (mac.toString('hex') !== json.crypto.mac) {
+      throw new Error('Key derivation failed - possibly wrong passphrase')
+    }
+
+    const decipher = crypto.createDecipheriv(
+      json.crypto.cipher,
+      derivedKey.slice(0, 16),
+      Buffer.from(json.crypto.cipherparams.iv, 'hex'),
+    )
+    const seed = runCipherBuffer(decipher, ciphertext)
+    return new Wallet(seed)
+  }
+
+  /*
+   * Based on https://github.com/ethereum/pyethsaletool/blob/master/pyethsaletool.py
+   * JSON fields: encseed, ethaddr, btcaddr, email
+   */
+  public static fromEthSale(input: string | EthSaleKeystore, password: string): Wallet {
+    const json: EthSaleKeystore = typeof input === 'object' ? input : JSON.parse(input)
+
+    const encseed = Buffer.from(json.encseed, 'hex')
+
+    // key derivation
+    const derivedKey = crypto.pbkdf2Sync(password, password, 2000, 32, 'sha256').slice(0, 16)
+
+    // seed decoding (IV is first 16 bytes)
+    // NOTE: crypto (derived from openssl) when used with aes-*-cbc will handle PKCS#7 padding internally
+    //       see also http://stackoverflow.com/a/31614770/4964819
+    const decipher = crypto.createDecipheriv('aes-128-cbc', derivedKey, encseed.slice(0, 16))
+    const seed = runCipherBuffer(decipher, encseed.slice(16))
+
+    const wallet = new Wallet(ethUtil.keccak256(seed))
+    if (wallet.getAddress().toString('hex') !== json.ethaddr) {
+      throw new Error('Decoded key mismatch - possibly wrong passphrase')
+    }
+    return wallet
+  }
+
+  // private getters
+
+  private get pubKey(): Buffer {
```

Comment:
I have really little context about this library, but why are this and the following method private?